typescript/src/schema/proto.ts

```diff
@@ -1,9 +1,30 @@
 import * as plugin from '../../../pbjs-genfiles/plugin';
 import { CommentsMap } from './comments';
+import * as objectHash from 'object-hash';
+import { milliseconds } from '../util';
+
+const defaultNonIdempotentRetryCodesName = 'non_idempotent';
+const defaultNonIdempotentCodes: plugin.google.rpc.Code[] = [];
+const defaultIdempotentRetryCodesName = 'idempotent';
+const defaultIdempotentCodes = [
+  plugin.google.rpc.Code.DEADLINE_EXCEEDED,
+  plugin.google.rpc.Code.UNAVAILABLE,
+];
+const defaultParametersName = 'default';
+const defaultParameters = {
+  initial_retry_delay_millis: 100,
+  retry_delay_multiplier: 1.3,
+  max_retry_delay_millis: 60000,
+  // note: the following four parameters are unused but currently required by google-gax.
+  // setting them to some big safe default values.
```

Comment:
Why are they required by google-gax? and is this something that is standard across languages (if you know)?

This is more for my own edification. I assumed supplying a 0 value for these would be OK.

Reply:
They are technically not required by google-gax logic, but are listed as required in the TypeScript type [here](https://github.com/googleapis/gax-nodejs/blob/master/src/gax.ts#L578). It will require a gax release to remove them from there, and when it's done, I will remove them from the generator as well.

Reply:
Oh that makes sense, thanks.